app/models/spree/order_decorator.rb

```diff
@@ -289,7 +289,8 @@ def associate_customer
 
   def ensure_customer
     unless associate_customer
-      self.customer = Customer.create(enterprise: distributor, email: email_for_customer, user: user)
+      customer_name = bill_address.andand.full_name
+      self.customer = Customer.create(enterprise: distributor, email: email_for_customer, user: user, name: customer_name, bill_address: bill_address.andand.clone, ship_address: ship_address.andand.clone)
```

Comment:
This piece of code is also useful for the admin interface creating a new order for a customer. So I think maybe it is still good to leave it here.